### ISSUE TITLE
can't self-pointing with an arrow to a same node in version 11.1.0

### ISSUE THREAD

**Issue by FeddyLix17** [mermaid-js/mermaid#5820]
### Description

From the version 11.1.0 up to v11.1.0+bfc4abe (the latest at the moment i'm writing) when you try to self pointing with an arrow to the same node it get the following error

---

Unable to render rich display

Cannot read properties of undefined (reading 'node')

For more information, see https://docs.github.com/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-mermaid-diagrams

---

using any version before 11.1.0 doesn't reproduce this behaviour

### Steps to reproduce

reproduce the bug

```
graph LR

A --> A
```

normal behaviour

```
graph LR

A --> B
```

### Screenshots

![image](https://github.com/user-attachments/assets/248d4d92-effe-469a-8007-c3eccd47a88a)

**Comment by 0xcaffebabe** [comment]
StateDiagram has same problem

**Comment by razterizer** [comment]
Yep. I got the same problem when the Mermaid Chart plugin in VSCode was updated this morning.

**Comment by ashishjain0512** [comment]
We can reproduce this and are working on a fix for it in the upcoming release.
@knsv

**Comment by FeddyLix17** [comment]
@knsv huge thanks for the fix!

i just wanna let you know that now the arrows towards the same node are much wider (i don't know if it was intentional).

```mermaid
graph LR;
    A((A))--test -->A((A))

    B((B))-- another test-->B((B))

    C((C))-- text-->C((C))
    C((C))-- another text-->D((D))


```

**Comment by M-casado** [comment]
Hi there @knsv, thanks for the fix as well! Now diagrams with self-references render, but I encounter similar issues as @FeddyLix17: the arrows for self-references are way off:
- Screenshot
![image](https://github.com/user-attachments/assets/2292ba9a-b25a-422a-b6a9-ef8fa9a2be3f)
````
    ZA -->|Enters periodic\nreview cycle| ZA
````
- Rendered in Mermaid.live
![image](https://github.com/user-attachments/assets/b10b78c9-6cbb-4689-8c31-1fa19b1fae05)

**Comment by Sun-ZhenXing** [comment]
This issue is still ongoing and is a problem with all 11.x releases.

10.x:

![image](https://github.com/user-attachments/assets/f1f1b3ad-0e0c-47a5-9bef-aa2772cb1c25)

But 11.x:

![image](https://github.com/user-attachments/assets/61813cae-bccd-4fdc-a77f-a5ec36866a2b)

```txt
stateDiagram-v2
  direction LR
  [*] --> 0
  0 --> 0 : a
  0 --> 1 : a
  0 --> 0 : b
  1 --> 2 : b
  2 --> (3) : b
```

That's not at all what I'm getting at.

**Comment by Inuart** [comment]
Yeah, all nodes with 2 or more self references are basically broken

**Comment by dkaszews** [comment]
Still an issue on live editor v11.6.0. Arrows look like the go through an invisible node two rows down, often in a really weird position.

**Comment by dkaszews** [comment]
Switching from `graph/flowchart` to `flowchart-elk` is a decent workaround, the self-references are rendered correctly, while the graphs themselves are IMO much nicer.
